data-plane/profiler/run.sh

```diff
@@ -0,0 +1,158 @@
+ASYNC_PROFILER_URL="https://github.com/jvm-profiling-tools/async-profiler/releases/download/v2.0-b1/async-profiler-2.0-b1-linux-x64.tar.gz"
+KAFKA_URL="https://downloads.apache.org/kafka/2.6.0/kafka_2.12-2.6.0.tgz"
+
+echo "Project root dir: ${PROJECT_ROOT_DIR}"
+echo "Resource dir: ${RESOURCES_DIR}"
+echo "Async profiler URL: ${ASYNC_PROFILER_URL}"
+echo "Kafka URL: ${KAFKA_URL}"
+
+# Build the data plane.
+cd "${PROJECT_ROOT_DIR}" && ./mvnw package -DskipTests -Dlicense.skip -Deditorconfig.skip --no-transfer-progress && cd - || exit 1
+
+# Download async profiler.
+rm async-profiler
+rm async-profiler.tgz
+mkdir async-profiler
+wget -O - ${ASYNC_PROFILER_URL} >async-profiler.tgz
+tar xzvf async-profiler.tgz -C async-profiler --strip-components=1
+
+# Download Apache Kafka.
+rm kafka
+rm kafka.tgz
+rm -r /tmp/kafka-logs/
+rm -r /tmp/zookeeper/
+mkdir kafka
+wget -O - ${KAFKA_URL} >kafka.tgz
+tar xzvf kafka.tgz -C kafka --strip-components=1
+
+ls -al
+
+# Start Zookeeper and Kafka.
+cd kafka || exit 1
+./bin/zookeeper-server-start.sh config/zookeeper.properties &
+zookeeper_pid=$!
+./bin/kafka-server-start.sh config/server.properties &
+kafka_pid=$!
+
+# Create our Kafka topic.
+./bin/kafka-topics.sh --create --topic attack-ingress-single --partitions 10 --bootstrap-server localhost:9092
+# Back to our previous dir
+cd ..
+
+# Define expected env variables.
+export PRODUCER_CONFIG_FILE_PATH=${RESOURCES_DIR}/config-kafka-broker-producer.properties
+export CONSUMER_CONFIG_FILE_PATH=${RESOURCES_DIR}/config-kafka-broker-consumer.properties
+export HTTPSERVER_CONFIG_FILE_PATH=${RESOURCES_DIR}/config-kafka-broker-httpserver.properties
+export WEBCLIENT_CONFIG_FILE_PATH=${RESOURCES_DIR}/config-kafka-broker-webclient.properties
+export DATA_PLANE_CONFIG_FILE_PATH=${RESOURCES_DIR}/ingress.json
+export CONFIG_TRACING_PATH=${RESOURCES_DIR}/tracing
+export LIVENESS_PROBE_PATH=/healthz
+export READINESS_PROBE_PATH=/readyz
+export METRICS_PATH=/metrics
+export METRICS_PUBLISH_QUANTILES="false"
+export EGRESSES_INITIAL_CAPACITY="1"
+
+# Define receiver specific env variables.
+export SERVICE_NAME="kafka-broker-receiver"
+export SERVICE_NAMESPACE="knative-eventing"
+export INGRESS_PORT="8080"
+export METRICS_PORT="9090"
+export INSTANCE_ID="receiver"
+
+# Run receiver.
+java \
+  -XX:+UnlockDiagnosticVMOptions \
+  -XX:+DebugNonSafepoints \
+  -Dlogback.configurationFile="${RESOURCES_DIR}"/config-logging.xml \
+  -jar "${PROJECT_ROOT_DIR}"/receiver/target/receiver-1.0-SNAPSHOT.jar &
+receiver_pid=$!
+
+# Define expected env variables.
+export SERVICE_NAME="kafka-broker-dispatcher"
+export SERVICE_NAMESPACE="knative-eventing"
+export INGRESS_PORT="8080"
+export METRICS_PORT="9089"
+export INSTANCE_ID="dispatcher"
+
+# Run dispatcher.
+java \
+  -XX:+UnlockDiagnosticVMOptions \
+  -XX:+DebugNonSafepoints \
+  -Dlogback.configurationFile="${RESOURCES_DIR}"/config-logging.xml \
+  -jar "${PROJECT_ROOT_DIR}"/dispatcher/target/dispatcher-1.0-SNAPSHOT.jar &
```

Comment:
In case you're still wondering the advantages of keeping the data plane independent from the control plane :smile:

Reply:
Never!